### PR TITLE
UI improvements in Dependencies dashboard & Packages explorer

### LINKED ISSUE #588: Update Dependency Dashboard
The current Dependency dashboard doesn't summarise the data well. A better design is required
Suggestions are experimented in [experiment/dependenciesViews](https://github.com/nexB/scancode-workbench/tree/experiment/dependenciesViews)

### PR DESCRIPTION
**Pull request by OmkarPh**
Updated charts:
image

Dependencies summary table:
image

Tabulated dependencies in the package view itself (Dependencies explorer on the left still works as before)
image

The packages view on the right didn't require much horizontal space, hence it was set to a default ~50% before
But, with the added table, it's changed to a default 65% (Suggest any other ratios, which suit better in most cases)
(It is resizable as usual)

### CODE REVIEW

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
@OmkarPh thanks++ updates look good!
A couple of nits for your consideration.

1. In the left main views section and the `view` in the bar above we have `Packages Explorer`, but once you're there the header is `Packages and Dependencies explorer`. We have to be consistent here and use only one. I'm leaning towards `Packages Explorer` here, but your call.
2. In the `Packages Explorer` when you click on one of the dependencies on the left, you have the details with the dependency attributes and their values. The you have their respective raw JSON there but it's titled `Raw package`, it should be `Raw dependency` instead?
3. In the dependency info dashboard, in the table of values by package type, we have a `No Value detected` there. I'm not sure we have PURLs without package types here, do you have an example of this? In any case let me check SCTK if there are actual instances of these. If not we can remove this row, and irrespective of that, we should only show this if there are values with no package type.
4. In the dependency info dashboard, in the table of values by package type, maybe we should have a title for the table like we have for other charts?
5. In the new dependencies table (and now that I notice it in the license matches table too similarly) we have empty space on the right in case of larger screens or even if you do full screen in smaller screens, while we still have some fields which are collapsed to a the second line. Do you think we can set these column values set such as we have minimum values for these, which will be there and then henceforth if it has more space than sum of the minimum of the column widths, we expand only some columns by proportions (while keeping smaller ones fixed). This is an enhancement and minor, if possible.

**Comment by mjherzog** [comment]
A minor point but names like Packages Explorer should be singular in most cases - e.g., Package Explorer

**Comment by OmkarPh** [comment]
> 1. In the left main views section and the `view` in the bar above we have `Packages Explorer`, but once you're there the header is `Packages and Dependencies explorer`. We have to be consistent here and use only one. I'm leaning towards `Packages Explorer` here, but your call.

Yep, having only `Package Explorer` makes sense.


> 2. In the `Packages Explorer` when you click on one of the dependencies on the left, you have the details with the dependency attributes and their values. Then you have their respective raw JSON there but it's titled `Raw package`, it should be `Raw dependency` instead?

My bad, it has to be `Raw dependency`

**Comment by OmkarPh** [comment]
> 4. In the dependency info dashboard, in the table of values by package type, maybe we should have a title for the table like we have for other charts?

I don't have a good title on top of my head, but what would be a good word to collectively represent `Runtime`, `Optional` & `Resolved`

**Comment by mjherzog** [comment]
The name that we usually use for Runtime, Optional etc. Dependency Scope.

**Comment by OmkarPh** [comment]
> 3. In the dependency info dashboard, in the table of values by package type, we have a `No Value detected` there. I'm not sure we have PURLs without package types here, do you have an example of this? In any case let me check SCTK if there are actual instances of these. If not we can remove this row, and irrespective of that, we should only show this if there are values with no package type.

Actually, I didn't get this `No value detected` anywhere, Can you send me the particular scan, I'll have a look
For example,
image

**Comment by OmkarPh** [comment]
> The name that we usually use for Runtime, Optional etc. Dependency Scope.

I can think of some options:
* `Dependency Scope summary by Package Type`
* `Dependency Scope summary across Package Types`

**Comment by AyanSinhaMahapatra** [comment]
> Actually, I didn't get this No value detected anywhere, Can you send me the particular scan, I'll have a look
For example,

Sure, here's a scan that had this value, attaching a screenshot too.
[elasticsearch_v32rc4.json.txt](https://github.com/nexB/scancode-workbench/files/12299841/elasticsearch_v32rc4.json.txt)
no_value_detected

I checked other JSONs and this does not seem to be present in all of them.

Also just a nit, notice how the table has only a few rows but takes up a lot more space than necessary, we should probably reduce the table height when there's not enough rows?

**Comment by OmkarPh** [comment]
Hi @AyanSinhaMahapatra 
I've done the changes requested

For the table width & height, there are some other changes in other branches which need to be taken into account, hence it is tracked separately in #598

File: src/pages/Packages/miscInfo.ts [hunk 1]
```diff
@@ -1,43 +1,47 @@
...
-  repository_homepage_url: null,
-  repository_download_url: null,
-  api_data_url: null,
-  datafile_paths: [],
-  datasource_ids: [],
-  purl: null,
-};
\ No newline at end of file
+
+// Obtaining a misc package via function to prevent preserving the same object across rerenders
```
Comment (AyanSinhaMahapatra) [inline src/pages/Packages/miscInfo.ts:5]: Nice!

**Review by AyanSinhaMahapatra — APPROVED** [review]
LGTM!
Thanks++ @OmkarPh , feel free to merge this. 

> For the table width & height, there are some other changes in other branches which need to be taken into account, hence it is tracked separately in https://github.com/nexB/scancode-workbench/issues/598

Sure, that makes sense.

**Comment by OmkarPh** [comment]
Thanks @AyanSinhaMahapatra 
Merging this

**Pull request merged**